plugin/plugin_publish.go

```diff
@@ -0,0 +1,175 @@
+package plugin
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+const (
+	goos_linux   = "linux"
+	goos_windows = "windows"
+	goos_darwin  = "darwin"
+
+	goarch_amd64 = "amd64"
+	goarch_arm64 = "arm64"
+)
+
+type PackageType string
+
+const (
+	PackageTypeNative PackageType = "native"
+	PackageTypeDocker PackageType = "docker"
+)
+
+type BuildTarget struct {
+	OS   string `json:"os"`
+	Arch string `json:"arch"`
+}
+
+// manifest is the plugin.json file inside the dist directory. It is used by CloudQuery registry
+// to be able to publish the plugin with all the needed metadata.
+type Manifest struct {
+	Name             string        `json:"name"`
+	Version          string        `json:"version"`
+	Title            string        `json:"title"`
+	ShortDescription string        `json:"short_description"`
+	Description      string        `json:"description"`
+	Categories       []string      `json:"categories"`
+	Protocols        []int         `json:"protocols"`
+	SupportedTargets []BuildTarget `json:"supported_targets"`
+	PackageType      PackageType   `json:"package_type"`
+}
+
+var buildTargets = []BuildTarget{
+	{goos_linux, goarch_amd64},
+	{goos_windows, goarch_amd64},
+	{goos_darwin, goarch_amd64},
+	{goos_darwin, goarch_arm64},
+}
+
+func isDirectoryExist(path string) bool {
+	info, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return info.IsDir()
+}
+
+func (p *Plugin) writeTablesJson(ctx context.Context, dir string) error {
+	tables, err := p.Tables(ctx, TableOptions{
+		Tables: []string{"*"},
+	})
+	if err != nil {
+		return err
+	}
+	buffer := &bytes.Buffer{}
+	m := json.NewEncoder(buffer)
+	m.SetIndent("", "  ")
+	m.SetEscapeHTML(false)
+	err = m.Encode(tables)
+	if err != nil {
+		return err
+	}
+	outputPath := filepath.Join(dir, "tables.json")
+	return os.WriteFile(outputPath, buffer.Bytes(), 0644)
+}
+
+func (p *Plugin) writeManifest(ctx context.Context, dir string) error {
+	manifest := Manifest{
+		Name:             p.Name(),
+		Version:          p.Version(),
+		Title:            p.title,
+		ShortDescription: p.shortDescription,
+		Description:      p.description,
+		Categories:       p.categories,
+		Protocols:        []int{3},
```

Comment:
Small thing but it would be nice if we can rather import the supported protocols as a constant from the relevant package